app/services/mapping/goodsshipment/TransactionNatureCodeBuilder.scala

```diff
@@ -17,10 +17,22 @@
 package services.mapping.goodsshipment
 
 import forms.declaration.NatureOfTransaction
+import javax.inject.Inject
 import uk.gov.hmrc.http.cache.client.CacheMap
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment
 import wco.datamodel.wco.declaration_ds.dms._2.GoodsShipmentTransactionNatureCodeType
 
+class GoodsShipmentNatureOfTransactionBuilder @Inject()() {
+  def buildThenAdd(data: Option[NatureOfTransaction], goodsShipment: GoodsShipment) {
```

Comment:
Similar here, why have an argument of an `Option` (Java compiler would tell us off). Just dont call the builder if it's `None`

Reply:
ok will change